subsys/mgmt/ec_host_cmd/backends/ec_host_cmd_backend_shi_ite.c

```diff
@@ -184,18 +194,17 @@ static int shi_ite_periph_send(const struct device *dev,
 	/* Copy preamble */
 	memcpy(data->out_msg, out_preamble, sizeof(out_preamble));
 
-	/* Copy data */
-	memcpy(data->out_msg + sizeof(out_preamble), buf->buf, buf->len);
-
-	/* Append our past-end byte, which we reserved space for. */
+	/* Data to send are already at out_msg + sizeof(out_preamble).
+	 * Append our past-end byte, which we reserved space for.
+	 */
 	for (int i = 0; i < EC_SHI_PAST_END_LENGTH; i++) {
-		data->out_msg[sizeof(out_preamble) + buf->len + i] = EC_SHI_PAST_END;
+		data->out_msg[sizeof(out_preamble) + data->tx->len + i] = EC_SHI_PAST_END;
```

Comment:
Would it make sense to create a pointer as local variable instead of doing double dereference every loop?

Reply:
I don't think so. It would be duplicating available data.